vividus-plugin-web-app/src/main/java/org/vividus/steps/ui/web/PageSteps.java

```diff
@@ -354,6 +366,20 @@ private URI updateUrlWithUserInfoForRedirects(URI pageUrl)
         return pageUrl;
     }
 
+    private URI moveUserInfoCredentialsToHttpContext(URI pageUrl, HttpClientContext httpContext)
+    {
+        UserInfo userInfo = UriUtils.getUserInfo(pageUrl);
+        Credentials credentials = new UsernamePasswordCredentials(userInfo.getUser(),
+                userInfo.getPassword().toCharArray());
+        BasicScheme scheme = new BasicScheme();
+        scheme.initPreemptive(credentials);
+        AuthCache authCache = new BasicAuthCache();
+        HttpHost host = HttpHost.create(pageUrl);
+        authCache.put(host, scheme);
+        httpContext.setAuthCache(authCache);
+        return UriUtils.removeUserInfo(pageUrl);
+    }
```

Comment:
```suggestion
    private void addBasicAuthToContext(UserInfo userInfo, HttpClientContext httpContext)
    {
        Credentials credentials = new UsernamePasswordCredentials(userInfo.getUser(),
                userInfo.getPassword().toCharArray());
        BasicScheme scheme = new BasicScheme();
        scheme.initPreemptive(credentials);
        AuthCache authCache = new BasicAuthCache();
        HttpHost host = HttpHost.create(pageUrl);
        authCache.put(host, scheme);
        httpContext.setAuthCache(authCache);
    }
```

Reply:
Also does it make sense to use `org.apache.hc.client5.http.ContextBuilder#preemptiveBasicAuth`?

Reply:
> Also does it make sense to use `org.apache.hc.client5.http.ContextBuilder#preemptiveBasicAuth`?

Yep, this way looks pretty easier.